### ISSUE TITLE
Here we go again. YouTube updated something and broke newpipe

### ISSUE THREAD

**Issue by razorshiv** [TeamNewPipe/NewPipe#2009]
Sometime this afternoon, newpipe started crashing while trying to load or play videos.

The error is "Could not parse decrypt function."

Schabi we need you!!! :)

**Comment by megamaiku** [comment]
yep, seem to be happening for most videos, but not all. It's happening for music I want to listen to 😢 

Version: 0.14.2

**Comment by aario** [comment]
So... Seems like I'm not alone. Here is my crash log:
```
{
   "user_action": "requested stream",
   "request": "https:\/\/www.youtube.com\/watch?v=DYYtuKyMtY8",
   "content_language": "GB",
   "service": "YouTube",
   "package": "org.schabi.newpipe",
   "version": "0.14.2",
   "os": "Linux Android 7.1.2 - 25",
   "time": "2019-01-17 00:27",
   "exceptions": [
      "org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor$DecryptException: Could not parse decrypt function \n\tat org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.loadDecryptionCode(YoutubeStreamExtractor.java:729)\n\tat org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:604)\n\tat org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:51)\n\tat org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:60)\n\tat org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:56)\n\tat org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:109)\n\tat org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$5fJcha6Sq5APJBLdG6osaJby-mc.call(lambda)\n\tat io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)\n\tat io.reactivex.Single.subscribe(Single.java:3438)\n\tat io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)\n\tat io.reactivex.Single.subscribe(Single.java:3438)\n\tat io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)\n\tat io.reactivex.Maybe.subscribe(Maybe.java:4154)\n\tat io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)\n\tat io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)\n\tat io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)\n\tat io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14479)\n\tat io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)\n\tat io.reactivex.Maybe.subscribe(Maybe.java:4154)\n\tat io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)\n\tat io.reactivex.Single.subscribe(Single.java:3438)\n\tat io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)\n\tat io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:237)\n\tat java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)\n\tat java.lang.Thread.run(Thread.java:761)\nCaused by: org.schabi.newpipe.extractor.utils.Parser$RegexException: failed to find pattern \"(encodeURIComponent=function\\([a-zA-Z0-9_]+\\)\\{.+?\\})\n\tat org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:63)\n\tat org.schabi.newpipe.extractor.utils.Parser.matchGroup1(Parser.java:52)\n\tat org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.loadDecryptionCode(YoutubeStreamExtractor.java:713)\n\t... 30 more\n"
   ],
   "user_comment": ""
}
```

**Comment by AnotherLife** [comment]
Yeah and I can't even long-press and play the videos on popup or background :(

Haaaalp!!

**Comment by atlanntiss** [comment]
Yes, this actually is not working for me too.

**Comment by The-Demon-Prosecutor** [comment]
I would like to point out that this is an issue with NewPipeExtractor and not NewPipe. I have made an issue on NewPipeExtractor already. Hence, it would be appreciated if we move to [#133](https://github.com/TeamNewPipe/NewPipeExtractor/issues/133).

**Comment by unknowtech** [comment]
same too error

**Comment by AnotherLife** [comment]
Oh my god, I tried a video and it works now. Hah, what the heck?

**Comment by azmarco** [comment]
It means youtube just need to change a decrypt method and newpipe just cease to work?

**Comment by ed828a** [comment]
I have the same problem. Could newpipe fix  this "Could not parse decrypt function."?

**Comment by dotvirus** [comment]
https://github.com/TeamNewPipe/NewPipeExtractor/issues/133 Seems to be fixed

**Comment by AnotherLife** [comment]
Nice, so I guess there will be a new release later?

**Comment by SaLtYdOgZ** [comment]
Funny. As is my custom, I look before I leap in regards to updates. I installed newpipe in April (so I could listen to youtube music in my car while keeping my screen turned off to not drain the battery) so I was on my initial version installed in April as the latest release. I tried it right before I upgraded and had no issues. So whatever version was available as current in April was working as of about 2 hours ago. Now, just a guru crash.
